Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -0,0 +1,357 @@
+    IsLindelof (f '' s) := by
+  intro l lne _ ls
+  have : NeBot (l.comap f ⊓ 𝓟 s) :=
+    comap_inf_principal_neBot_of_image_mem lne (le_principal_iff.1 ls)
+  obtain ⟨x, hxs, hx⟩ : ∃ x ∈ s, ClusterPt x (l.comap f ⊓ 𝓟 s) := @hs _ this _ inf_le_right
+  haveI := hx.neBot
+  use f x, mem_image_of_mem f hxs
+  have : Tendsto f (𝓝 x ⊓ (comap f l ⊓ 𝓟 s)) (𝓝 (f x) ⊓ l) := by
+    convert (hf x hxs).inf (@tendsto_comap _ _ f l) using 1
+    rw [nhdsWithin]
+    ac_rfl
+  exact this.neBot
+
+/-- A continuous image of a Lindelöf set is a Lindelöf set within the codomain. -/
+theorem IsLindelof.image {f : X → Y} (hs : IsLindelof s) (hf : Continuous f) :
+    IsLindelof (f '' s) := hs.image_of_continuousOn hf.continuousOn
+
+/-- A filter with the countable intersection property that is finer than the principal filter on
+a Lindelöf set `s` contains any open set that contains all clusterpoints of `s`. -/
+theorem IsLindelof.adherence_nhdset {f : Filter X} [CountableInterFilter f] (hs : IsLindelof s)
+    (hf₂ : f ≤ 𝓟 s) (ht₁ : IsOpen t) (ht₂ : ∀ x ∈ s, ClusterPt x f → x ∈ t) : t ∈ f :=
+  Classical.by_cases mem_of_eq_bot fun (this : f ⊓ 𝓟 tᶜ ≠ ⊥) ↦
+  have hinf : CountableInterFilter (f ⊓ 𝓟 tᶜ) := countableInterFilter_inf _ _
+  let ⟨x, hx, (hfx : ClusterPt x <| f ⊓ 𝓟 tᶜ)⟩ := @hs _ ⟨this⟩ hinf <| inf_le_of_left_le hf₂
+  have : x ∈ t := ht₂ x hx hfx.of_inf_left
+  have : tᶜ ∩ t ∈ 𝓝[tᶜ] x := inter_mem_nhdsWithin _ (ht₁.mem_nhds this)
+  have A : 𝓝[tᶜ] x = ⊥ := empty_mem_iff_bot.1 <| compl_inter_self t ▸ this
+  have : 𝓝[tᶜ] x ≠ ⊥ := hfx.of_inf_right.ne
+  absurd A this
+
+/--For every open cover of a Lindelöf set, there exists a countable subcover. -/
+theorem IsLindelof.elim_countable_subcover {ι : Type v} (hs : IsLindelof s) (U : ι → Set X)
+    (hUo : ∀ i, IsOpen (U i)) (hsU : s ⊆ ⋃ i, U i) :
+    ∃ r : Set ι, r.Countable ∧ (s ⊆ ⋃ i ∈ r, U i) := by
+  have he : ∃ r : Set ι, r.Countable ∧ ∅ ⊆ ⋃ i ∈ r, U i := by use ∅; simp
+  have hmono : ∀ ⦃s t : Set X⦄, s ⊆ t → (∃ r : Set ι, r.Countable ∧ t ⊆ ⋃ i ∈ r, U i)
+      → (∃ r : Set ι, r.Countable ∧ s ⊆ ⋃ i ∈ r, U i) := by
+    intro _ _ hst ⟨r, ⟨hrcountable,hsub⟩⟩
+    exact ⟨r,hrcountable,Subset.trans hst hsub⟩
+  have hcountable_union : ∀ (S : Set (Set X)), S.Countable
+      → (∀ s ∈ S, ∃ r : Set ι, r.Countable ∧ (s ⊆ ⋃ i ∈ r, U i))
+      → ∃ r : Set ι, r.Countable ∧ (⋃ s ∈ S, s ⊆ ⋃ i ∈ r, U i) := by
+    intro S hS hsr
+    choose! r hr using hsr
+    refine ⟨⋃ s ∈ S, r s, hS.biUnion_iff.mpr (fun s hs ↦ (hr s hs).1), ?_⟩
+    refine iUnion₂_subset ?h.right.h
+    simp only [mem_iUnion, exists_prop, iUnion_exists, biUnion_and']
+    exact fun i is x hx ↦ mem_biUnion is ((hr i is).2 hx)
+  have h_nhds : ∀ x ∈ s, ∃ t ∈ 𝓝[s] x, ∃ r : Set ι, r.Countable ∧ (t ⊆ ⋃ i ∈ r, U i) := by
+    intro x hx
+    let ⟨i, hi⟩ := mem_iUnion.1 (hsU hx)
+    refine ⟨U i, mem_nhdsWithin_of_mem_nhds ((hUo i).mem_nhds hi),{i}, ?_⟩
+    refine ⟨by simp, ?_⟩
```

Comment:
Some more general advice on mathlib style: if you just have two subgoals, I think "constructor" might be more readable. In this case, you can combine these into one line.
`refine ⟨U i, mem_nhdsWithin_of_mem_nhds ((hUo i).mem_nhds hi), {i}, by simp, ?_⟩`